File: examples/sample-game/assets/scripts/Spinner.cpp
#include "Spinner.hpp"

void Spinner::start(epok::Transform&) {}

void Spinner::update(epok::Transform& transform, epok::Fixed dt) {
    transform.rotation[1] += speed * dt;
    if (transform.rotation[1] >= 360.0) transform.rotation[1] -= 360.0;
    if (transform.rotation[1] < 0.0) transform.rotation[1] += 360.0;
}

File: templates/Spinner.lua
---@class Spinner : epok.ActorComponent
local Spinner = epok.ActorComponent:extend()

-- The smallest possible behaviour: turn the actor this component is attached to.
-- Degrees per second, editable per instance from the Inspector.
Spinner.speed = 45.0

---@override
function Spinner:tick(delta_seconds)
    self.rotation.y = self.rotation.y + self.speed * delta_seconds
end

return Spinner
